proposals/field-keyword-nullability.md

```diff
@@ -0,0 +1,88 @@
+# `field` keyword nullability
+
+## Goals
```

Comment:
Consider a motivation section with an example of what a user migjht want to write that we think is very reasonable.  (or add that to this section).

Reply:
I think Joseph had some great samples here, let's use that and I'll make a revision to put it all in the field-keyword.md file.